compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDmaUtils.cpp

```diff
@@ -162,10 +208,18 @@ LogicalResult combineAccessPatterns(RewriterBase &rewriter,
     newOffsets = offsetsB;
     newSizes = sizesB;
     newStrides = stridesB;
-    for (int i = 1; i <= offsetsA.size(); i++) {
-      if (offsetsA[offsetsA.size() - i] != offsetsB[offsetsB.size() - i]) {
-        newOffsets[newOffsets.size() - i] = offsetsA[offsetsA.size() - i];
-        break;
+    // If the offset on the first dimension of B is larger than zero, we can
+    // just decrease that one to accomplish the access pattern merge. Otherwise,
```

Comment:
nit ?
```suggestion
    // just decrease that by one to accomplish the access pattern merge. Otherwise,
```
(as I see `rewriter.getI64IntegerAttr(offset.value() - 1);` below)